File: 141-linked-list-cycle/linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        HashSet<ListNode> hset = new HashSet<>();
        ListNode current = head;
        while(current!=null){
            if(hset.contains(current)){
                return true;
            }
            hset.add(current);
            current= current.next;
        }
        return false;
    }
}

File: 141-linked-list-cycle/linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    const set = new Set();
    let current = head;
    while(current!=null){
        if(set.has(current)){
            return true;
        }
        set.add(current);
        current = current.next;
    }

    return false;
    
    
};